resolwe/flow/tests/test_filtering.py

```diff
@@ -0,0 +1,233 @@
+            name='Test collection 1',
+            slug='test-collection-1',
+            descriptor_schema=cls.descriptor_schema_1,
+            contributor=cls.user_1
+        )
+        cls.collection_1.created = datetime.datetime(2016, 7, 30, 13, 59)
+        cls.collection_1.save()
+
+        cls.collection_2 = Collection.objects.create(
+            name='Test collection 2',
+            slug='test-collection-2',
+            descriptor_schema=cls.descriptor_schema_1,
+            contributor=cls.user_1
+        )
+        cls.collection_2.created = datetime.datetime(2016, 8, 30, 14, 59)
+        cls.collection_2.save()
+
+        cls.collection_3 = Collection.objects.create(
+            name='Another collection',
+            slug='another-collection',
+            descriptor_schema=cls.descriptor_schema_2,
+            contributor=cls.user_2
+        )
+        cls.collection_3.created = datetime.datetime(2013, 1, 15, 7, 59)
+        cls.collection_3.save()
+
+        cls.proc = Process.objects.create(
+            name='Test process',
+            slug='test-process',
+            contributor=cls.user_1,
+            type='data:test:'
+        )
+
+        cls.data = Data.objects.create(
+            name='Test data',
+            slug='test-data',
+            contributor=cls.user_1,
+            process=cls.proc,
+            status=Data.STATUS_DONE
+        )
+        cls.collection_1.data.add(cls.data)
+        cls.collection_2.data.add(cls.data)
+
+    def _apply_filter(self, filters, expected):
+        filtered = CollectionFilter(filters, queryset=Collection.objects.all())
+        six.assertCountEqual(self, filtered, expected)
+
+    def test_filter_id(self):
+        self._apply_filter({'id': self.collection_1.pk}, [self.collection_1])
+        self._apply_filter({'id__in': '{},{}'.format(self.collection_1.pk, self.collection_2.pk)},
+                           [self.collection_1, self.collection_2])
+
+    def test_filter_slug(self):
+        self._apply_filter({'slug': 'test-collection-2'}, [self.collection_2])
+        self._apply_filter({'slug__startswith': 'test'}, [self.collection_1, self.collection_2])
+
+    def test_filter_name(self):
+        self._apply_filter({'name': 'Test collection 2'}, [self.collection_2])
+        self._apply_filter({'name__startswith': 'Test'}, [self.collection_1, self.collection_2])
+
+    def test_filter_contributor(self):
+        self._apply_filter({'contributor': self.user_1.pk},
+                           [self.collection_1, self.collection_2])
+
+    def test_filter_created(self):
+        self._apply_filter({'created__year': '2016'}, [self.collection_1, self.collection_2])
+        self._apply_filter({'created__year__gt': '2015'}, [self.collection_1, self.collection_2])
+        self._apply_filter({'created__year__gt': '2015', 'created__month__gte': '8'}, [self.collection_2])
+
+    def test_filter_modified(self):
+        pass  # XXX: `modified` is changed on save
```

Comment:
You can still count the number of objects that you have created until now and that you have not created anything from now on.